### PR TITLE
Amend the Dart yaml file support to include pubspec.yaml files and fix_data.yaml files

### PR DESCRIPTION
**Pull request by jwren**
@alexander-doroshko

### CODE REVIEW

File: Dart/resources/META-INF/dart-yaml.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
                      enabledByDefault="true" level="WARNING" language="yaml"
                      implementationClass="com.jetbrains.lang.dart.ide.inspections.DartPathPackageReferenceInspection"/>
     <psi.referenceContributor language="yaml" implementation="com.jetbrains.lang.dart.psi.PubspecYamlReferenceContributor" order="last"/>
-    <fileType name="YAML" fileNames=".analysis_options"/>
+    <fileType name="YAML" fileNames=".analysis_options;analysis_options.yaml;pubspec.yaml;fix_data.yaml"/>
```
Comment (alexander-doroshko) [inline Dart/resources/META-INF/dart-yaml.xml:8]: I don' think we need to list `*.yaml` files here, they already match standard `*.yaml` file pattern. Have you faced any issues without this change?

Comment (jwren) [reply Dart/resources/META-INF/dart-yaml.xml]: No... doing a double-take after your comment, PR updated.

**Comment by alexander-doroshko** [comment]
@jwren Curious, should code completion already work in these files or this change is for future?

I don't have code completion in `analysis_options.yaml`, `pubspec.yaml` or `fix_data.yaml` using SDK 2.10.0-110.3.beta. 
Neither have I errors in `pubspec.yaml` or `fix_data.yaml`. 

No code completion in  `analysis_options.yaml` might be a Windows-specific issue. Do you have it working on Mac/Linux? There's a long-living bug report about that: https://github.com/dart-lang/sdk/issues/25114

**Comment by jwren** [comment]
Future -- trying to motivate some work in the analysis server, but I want to be able to say that contributions will be able to get through to users.

FYI, we do have pubspec.yaml completions via https://www.reddit.com/r/FlutterDev/comments/doqk6n/there_is_now_a_json_schema_for_pubspecyaml_what/

**Comment by alexander-doroshko** [comment]
Ok, thanks, merged.
True, we have code completion in pubspec, I mean it's not from Analysis Server.

**Pull request closed without merging**